### DISCUSSION TITLE
user domain links to several IdP

### DISCUSSION THREAD

**Discussion by sgrannec** [sventorben/keycloak-home-idp-discovery#105] [Ideas / New features]
Hello Sven-Torben Janus,

Thanks a lot for your Keycloak's feature ! It is very usefull !

I have a usecase : a user (email domain = @domain.com) could connect to several IdPs : for example, the same user could have access to severals external IdPs and internal IdPs.
In fact when the user tries to log in, he enters his address mail. Then as his domain is links to several IdPs, instead of being redirected to one IdP Web page (In fact now a unique domain is configured by IdP), he is redirected to another Web page that lists all IdPs linked to his email domain.

What do you think about it ? is-it possible to implement it ?

Thanks,
Best regards.

**Comment by sventorben** [comment]
Hello @sgrannec,

thanks for your suggestion.
If I understand this correctly, a user ends up with a list of identity providers to choose from, right?
That is supported by Keycloak out of the box if you configure multiple IdPs, because it is common with social providers.

I do not understand how the domain comes into play here. Do you want to filter the list of IdPs based on the domain?

Best regards,

Sven-Torben

**Reply by sgrannec** [reply]
Hello,
Yes of course, Keycloak provides this features but you see all IdPs configured. What I want is to filter the list of IdPs based on the domain.

Best regards.

**Reply by sventorben** [reply]
@sgrannec check the latest version ;)
